virtio-devices/src/block.rs

```diff
@@ -690,9 +690,19 @@ impl Block {
                 (disk_nsectors, avail_features, 0, config, false)
             };
 
-        let serial = serial
-            .map(Vec::from)
-            .unwrap_or_else(|| build_serial(&disk_path));
+        let serial = if let Some(s) = serial {
```

Comment:
Looking at this PR again - I think the commit message is a bit confusing. Something like:

```
virtio-devics: block: Truncate device serial to maximum supported length

Truncate the device serial to the maximum supported length to ensure that it is correctly matched during guest boot disk discovery.
```